api/v1alpha1/types.go

```diff
@@ -0,0 +1,55 @@
+package v1alpha1
+
+import (
+	capi "github.com/hashicorp/consul/api"
+)
+
+type MeshGatewayMode string
+
+const (
+	// MeshGatewayModeDefault represents no specific mode and should
+	// be used to indicate that a different layer of the configuration
+	// chain should take precedence
+	MeshGatewayModeDefault MeshGatewayMode = ""
+
+	// MeshGatewayModeNone represents that the Upstream Connect connections
+	// should be direct and not flow through a mesh gateway.
+	MeshGatewayModeNone MeshGatewayMode = "none"
+
+	// MeshGatewayModeLocal represents that the Upstrea Connect connections
+	// should be made to a mesh gateway in the local datacenter. This is
+	MeshGatewayModeLocal MeshGatewayMode = "local"
+
+	// MeshGatewayModeRemote represents that the Upstream Connect connections
+	// should be made to a mesh gateway in a remote datacenter.
+	MeshGatewayModeRemote MeshGatewayMode = "remote"
+)
+
+// MeshGatewayConfig controls how Mesh Gateways are used for upstream Connect
+// services
+type MeshGatewayConfig struct {
+	// Mode is the mode that should be used for the upstream connection.
+	Mode string `json:"mode,omitempty"`
+}
+
+//ToConsul returns the MeshGatewayConfig for the entry
+func (m MeshGatewayConfig) ToConsul() capi.MeshGatewayConfig {
+	switch m.Mode {
```

Comment:
should we make that change we talked about?

Reply:
good catch